plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/model/JDK.java

```diff
@@ -49,16 +46,25 @@ public enum JDK {
      */
     private final String maximumCoreVersion;
 
+    /**
+     * The latest compatible version of jenkins-test-harness version
+     */
+    private final String latestTestHarnessVersion;
+
     /**
      * Constructor
      * @param major The Java major version
      * @param lts If the version is LTS
+     * @param compatibleSince The compatibility since for this JDK
+     * @param maximumCoreVersion The maximum required core version
+     * @param latestTestHarnessVersion The latest compatible version of jenkins-test-harness
      */
-    JDK(int major, boolean lts, String compatibleSince, String maximumCoreVersion) {
+    JDK(int major, boolean lts, String compatibleSince, String maximumCoreVersion, String latestTestHarnessVersion) {
```

Comment:
Note for myself (out of context of this PR). This is worth adding also the parent. For example 4.88 is Java 11 and 5.x is Java 17.

It would avoid harcoding the value in several places (versions.properties or recipes) and have it centralised here.

There is some strict relation with Jenkins version and supported parent pom.xml